### ISSUE TITLE
read_json/auto with SELECT output as source parameter raises INTERNAL Error: Attempted to access index 0 within vector of size 0

### ISSUE THREAD

**Issue by woozyking** [duckdb/duckdb#7599]
### What happens?

When `read_json` or `read_json_auto` takes a SELECT output as its source parameter raises `INTERNAL Error: Attempted to access index 0 within vector of size 0`

### To Reproduce

Using a CTE:

```sql
WITH urls AS (
    SELECT url
    FROM read_csv_auto('https://minedbuildings.blob.core.windows.net/global-buildings/dataset-links.csv', header=True)
    WHERE location IN ('Canada', 'UnitedStates')
)
SELECT *
FROM read_json_auto((SELECT url FROM urls LIMIT 3), format=newline_delimited)
WHERE properties.height > -1.0
LIMIT 10
```

Which would yield `Error: INTERNAL Error: Attempted to access index 0 within vector of size 0`

A dedicated table:

```sql
CREATE TABLE urls
AS SELECT url
FROM read_csv_auto('https://minedbuildings.blob.core.windows.net/global-buildings/dataset-links.csv', header=True)
WHERE location IN ('Canada', 'UnitedStates');

SELECT *
FROM read_json_auto((SELECT url FROM urls LIMIT 3), format=newline_delimited)
WHERE properties.height > -1.0
LIMIT 10;
```

or subquery would be the same:

```sql
SELECT *
FROM read_json_auto((
    SELECT url
    FROM read_csv_auto('https://minedbuildings.blob.core.windows.net/global-buildings/dataset-links.csv', header=True)
    WHERE location IN ('Canada', 'UnitedStates')
    LIMIT 3
), format=newline_delimited)
WHERE properties.height > -1.0
LIMIT 10;
```

### OS:

macOS

### DuckDB Version:

v0.8.0 e8e4cea5ec

### DuckDB Client:

CLI

### Full Name:

Runzhou Li

### Affiliation:

mation

### Have you tried this on the latest `master` branch?

- [ ] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #7603 that clears up the error message here.

This is expected not to work - functions like `read_json_auto` are special functions as their output names and types *depend on the input parameters*. As such, their input must be known at query compile time - i.e. the input must be constant parameters.

**Comment by woozyking** [comment]
> Thanks for the report! I have pushed a fix in #7603 that clears up the error message here.
> 
> 
> 
> This is expected not to work - functions like `read_json_auto` are special functions as their output names and types *depend on the input parameters*. As such, their input must be known at query compile time - i.e. the input must be constant parameters.
> 
> 

Thanks for the prompt response.

What would be the most optimal way of doing something like this, aside from a naive exchange between the DB and a client language such as Python? I currently have something like:

```Python
urls = [
    r[0] for r in duckdb.sql("""
        query that obtains needed urls
    """).fetchall()
]

duckdb.sql(f"""
  query that plugs in constant {urls}
""")
```
